backend/src/test/java/com/redhat/cloud/notifications/db/repositories/BehaviorGroupRepositoryTest.java

```diff
@@ -463,6 +463,204 @@ void testNotAllowedCreateMoreBehaviorGroups() {
+
+        final Application application = this.resourceHelpers.createApplication(bundle.getId());
+        final EventType eventType = this.resourceHelpers.createEventType(application.getId(), "name", "display-name", "description");
+
+        // Call the function under test.
+        final NotFoundException ex = Assertions.assertThrows(
+            NotFoundException.class,
+            () -> this.behaviorGroupRepository.deleteBehaviorGroupFromEventType(eventType.getId(), behaviorGroup.getId(), DEFAULT_ORG_ID)
+        );
+
+        Assertions.assertEquals("the specified behavior group was not found for the given event type", ex.getMessage(), "unexpected exception message");
+    }
+
+    /**
+     * Tests that when a {@link BehaviorGroup} is removed from the behavior gorups collection of an @{@link EventType},
+     * no error is raised.
+     */
+    @Test
+    void testDeleteBehaviorGroupFromEventType() {
+        final Bundle bundle = this.resourceHelpers.createBundle();
+        final BehaviorGroup behaviorGroup = this.resourceHelpers.createBehaviorGroup(DEFAULT_ACCOUNT_ID, DEFAULT_ORG_ID, "display-name", bundle.getId());
+
+        final Application application = this.resourceHelpers.createApplication(bundle.getId());
+        final EventType eventType = this.resourceHelpers.createEventType(application.getId(), "name", "display-name", "description");
+
+        // First insert the behavior group as part of the even type's behavior groups.
+        this.behaviorGroupRepository.updateEventTypeBehaviors(DEFAULT_ORG_ID, eventType.getId(), Set.of(behaviorGroup.getId()));
+
+        // Then call the function under test.
+        this.behaviorGroupRepository.deleteBehaviorGroupFromEventType(eventType.getId(), behaviorGroup.getId(), DEFAULT_ORG_ID);
+    }
+
+    /**
+     * Tests that when a non-existing UUID is given for a behavior group, a
+     * {@link NotFoundException} is raised with the proper error message.
+     */
+    @Test
+    void testAppendBehaviorGroupEventTypeNotExists() {
+        final NotFoundException exception = Assertions.assertThrows(
+            NotFoundException.class,
+            () -> this.behaviorGroupRepository.appendBehaviorGroupToEventType(
+                UUID.randomUUID().toString(),
+                UUID.randomUUID(),
+                UUID.randomUUID()
+            )
+        );
+
+        Assertions.assertEquals("the specified behavior group doesn't exist or the specified event type doesn't belong to the same bundle as the behavior group", exception.getMessage(), "unexpected exception received for a non existing behavior group");
+    }
+
+    /**
+     * Tests that when an existing UUID is given for a behavior group, but with
+     * a different organization ID with the one expected, which simulates a
+     * different tenant, a {@link NotFoundException} is raised with the proper
+     * error message.
+     */
+    @Test
+    void testAppendBehaviorGroupEventTypeNotExistsOrgId() {
```

Comment:
After the query change, there's no way to tell which of the following factors causes the test failure:
- BG from a different org
- unknown event type (because you're passing a random UUID to the tested method)

Reply:
I have modified the test to provide a valid behavior group, so that in this case we can be "more sure" that the error is due to providing a random event type...